### PR TITLE
#170 Restore contract with placeholder implemented + tested.

### PR DESCRIPTION
**Pull request by criske**
PR for #155 

- left puzzle to use the real contract restore api whan available.

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

File: src/main/java/com/selfxdsd/selfweb/api/ContractsApi.java [hunk 1]
```diff
@@ -329,4 +366,51 @@ public ContractsApi(final User user, final Self self) {
...
+     * Restores a Contract marked for removal.
+     * @param owner Owner of the project (username or org name).
+     * @param name Simple name of the project.
+     * @param username Contributor's username.
+     * @param role Contributor's role.
+     * @return JsonResponse.
+     * @checkstyle ParameterNumber (10 lines)
+     */
+    @PutMapping(
+        value = "/projects/{owner}/{name}/contracts/{username}/restore",
```
Comment (criske) [inline src/main/java/com/selfxdsd/selfweb/api/ContractsApi.java:380]: @amihaiemil Wouldn't be better to use here `@DeleteMapping` with `/projects/{owner}/{name}/contracts/{username}/mark` and for mark for removal endpoint  `@PutMapping`? This way we'll share the same endpoint.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2020-11-17T10:50:31.524953``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: src/main/java/com/selfxdsd/selfweb/api/ContractsApi.java (hunk 1 above)
Comment (amihaiemil) [reply src/main/java/com/selfxdsd/selfweb/api/ContractsApi.java:380]: @criske Why use DeleteMapping when this endpoint restores a contract? I think Put is the correct way :-?

**Review by amihaiemil — COMMENTED** [review]
@criske I don't understand your question :D

**Comment by criske** [comment]
> @criske I don't understand your question :D

@amihaiemil  I was thinking:
- PUT `/projects/{owner}/{name}/contracts/{username}/mark` - marks the contract for removal, but the contract is not deleted yet, needs 30+.
would read as: **_"put a mark for removal for contract etc..."_**
- DELETE `/projects/{owner}/{name}/contracts/{username}/mark` - deletes that mark, meaning that contract is restored.
would read as: _**"remove the mark for removal for contract etc..,"**_

**Comment by amihaiemil** [comment]
@criske right, I guess it makes sense. 

It's a good idea, but leave a todo for this, don't do the change here :)

**Comment by amihaiemil** [comment]
@rultor merge

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**